src/Nut/ConfigSet.php

```diff
@@ -36,10 +36,11 @@ protected function doExecute(YamlUpdater $updater, InputInterface $input, Output
         $value = $input->getArgument('value');
         $backup = $input->getOption('backup');
 
-        $updater->change($key, $value, $backup, true);
-        if (is_bool($value)) {
-            $value = $value ? 'true' : 'false';
-        }
+        $newValue = $value === 'true' || $value === 'false'
+            ? filter_var($value, FILTER_VALIDATE_BOOLEAN)
+            : $value
+        ;
```

Comment:
I don't like this. Assignment, comparisons and a ternary operator in one statement.. Let's keep it separate: 

```
if ($value === 'true' || $value === 'false') {
    $value = filter_var($value, FILTER_VALIDATE_BOOLEAN);
}
$updater->change($key, $value, $backup, true);
```

Reply:
You mean like in a large chunk of the rest of the code base.

The thing here is that then it has to become an if/else, or alternatively have an additional if/else on return to switch the boolean back to text for the output

Reply:
Right, I missed the part about needing to keep the original `$value`. Yeah, let's not do that. :-)

Reply:
Funnily enough it was what took me so long today to get this one to PR … it just seemed the least evil option :disappointed: